Missing mesh page + Mesh Manager pages (#1837)
Kong/developer.konghq.com: app/mesh/mesh-manager/migrate-zone.md

---
title: Migrate a self-managed zone Control Plane to {{site.konnect_short_name}}
content_type: reference
description: 'Move your existing {{site.mesh_product_name}} zone Control Planes from a self-managed global Control Plane to a managed global Control Plane in {{site.konnect_short_name}}.'
layout: reference

> permalink: /mesh-manager/migrate-zone/

Reviewer: ```suggestion
```

* **Setup wizard:** A UI-driven setup guides you through zone configuration.

This guide explains how to migrate a self-managed zone Control Plane to {{site.konnect_short_name}}.

![Before migration](/assets/images/konnect/diagram-mesh-migration-before.png)
> **Figure 1:** {{site.mesh_product_name}} services and data plane proxies connect to a self-managed global Control Plane.

![After migration](/assets/images/konnect/diagram-mesh-migration-after.png)
> **Figure 2:** After migration, the global Control Plane is hosted in {{site.konnect_short_name}}, while zones and services remain self-managed.

## Limitation

This process assumes you're migrating zones one by one. During migration, zone-to-zone communication may break temporarily because each zone's Zone Ingress must be registered with the new global Control Plane in {{site.konnect_short_name}}. Until both zones are migrated, cross-zone service discovery won't work.

## Prerequisites

* A Kubernetes or universal cluster with a zone Control Plane connected to a self-managed global Control Plane

> * [`kumactl`](/mesh/latest/production/install-kumactl/) installed and configured

Reviewer: ```suggestion
* [`kumactl`](/mesh/#install-kong-mesh) installed and configured
```

   meshfaultinjections meshgateways meshgatewayroutes meshhealthchecks meshhttproutes meshloadbalancingstrategies
   meshproxypatches meshratelimits meshretries meshtcproutes meshtimeouts meshtraces meshtrafficpermissions
   proxytemplates rate-limits retries timeouts traffic-logs traffic-permissions traffic-routes traffic-traces
   virtual-outbounds access-role-bindings access-roles accessaudits meshglobalratelimits meshopas opa-policies"

   meshes=$(kumactl get meshes -oyaml | yq '.items[].name')

   for mesh in ${meshes}; do
     kumactl get mesh ${mesh} -oyaml | yq '(del(.creationTime,.modificationTime))' > "${outdir}/${mesh}.yaml"
     echo "---" >> "${outdir}/${mesh}.yaml"

     for type in ${types}; do
       kumactl get ${type} --mesh ${mesh} -oyaml | yq '.items[] |= (del(.creationTime,.modificationTime)) | .items[] | split_doc' | grep ^ >> "${outdir}/${mesh}.yaml" && echo "---" >> "${outdir}/${mesh}.yaml"
     done
   done
   ```

1. If mTLS is enabled, copy mesh secrets:

   * **Builtin CA:** Copy secrets named `{mesh}.ca-builtin-cert-{backend}` and `{mesh}.ca-builtin-key-{backend}`.
   * **Provided CA:** Copy the secrets listed in the mesh resource.

>    See [mTLS secret storage](/mesh/latest/policies/mutual-tls/#storage-of-secrets) for details.

Reviewer: ```suggestion
   See [mTLS secret storage](/mesh/policies/meshtls/#storage-of-secrets) for details.
```

1. Switch to the {{site.konnect_short_name}} global Control Plane:

   ```bash
   kumactl config control-planes switch --name {konnect-config-name}
   ```

1. Apply the exported resources:

   ```bash
   kumactl apply -f {file_name}
   ```

## Connect the zone Control Plane to {{site.konnect_short_name}}

> 1. [Create a new zone](/konnect/mesh-manager/service-mesh/#create-a-zone-in-the-global-control-plane) in {{site.konnect_short_name}}.

Reviewer: ```suggestion
1. [Create a new zone](/mesh-manager/service-mesh/#create-a-zone-in-the-global-control-plane) in {{site.konnect_short_name}}.

```
